crates/pessimistic-proof/src/local_balance_tree.rs

```diff
@@ -38,3 +43,22 @@ impl From<LocalBalanceTree> for pessimistic_proof_core::local_balance_tree::Loca
         Self { root: tree.root }
     }
 }
+
+pub struct BalanceTree(pub Smt<LOCAL_BALANCE_TREE_DEPTH>);
+
+impl BalanceTree {
+    /// Returns all the non-zero token balance contained in this balance tree.
+    pub fn get_all_balances(
+        &self,
+    ) -> Result<Vec<(SmtPath<LOCAL_BALANCE_TREE_DEPTH>, Digest)>, SmtError> {
+        Ok(self.0.entries()?.into_iter().collect())
+    }
```

Comment:
Could this return an iterator instead of creating an intermediate Vec? The caller may still decide to `collect`.